core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -777,16 +782,18 @@ public Builder addSchemas(final URI... schemaURI) {
         }
 
         public Builder addSchema(final Path schemaPath) {
-            if (schemaPath != null) {
-                try (DirectoryStream<Path> stream = Files.isDirectory(schemaPath) ? Files.newDirectoryStream(schemaPath) : null) {
-                    if (stream != null) {
-                        stream.forEach(this::addSchema);
-                    } else {
-                        addSchema(Files.readAllBytes(schemaPath));
+            try {
+                // If directory load all schemas from it
+                if (Files.isDirectory(schemaPath)) {
```

Comment:
I'm not quite sure what you mean the stream is not created until the next line where its in a try with resources block? The `readAllBytes()`method handles the close automatically see [Javadocs](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#readAllBytes(java.nio.file.Path))

Reply:
I don't see any issue with that. While this refactor is better than before, I think it may be possible to improve further and not have a nested `try`.

Reply:
The only way to not have the try block and still use a directory stream would be to have a finally block in the outer try, unsure which is more readable?